libraries/botbuilder-ai/src/languageTranslator.ts

```diff
@@ -29,20 +31,33 @@ export interface TranslationContext {
  * get a translated experience, and also a LUIS model allowing the user to ask to speak a language.
  */
 export class LanguageTranslator implements Middleware {
-    private luisClient: LuisClient;
     private translator: Translator;
+    private getUserLanguage: ((context: BotContext) => string) | undefined;
+    private setUserLanguage: ((context: BotContext) => Promise<boolean>) | undefined;
 
-    public constructor(translatorKey: string, protected nativeLanguages: string[], protected luisAppId: string, protected luisAccessKey: string) {
-        this.luisClient = new LuisClient();
-        this.translator = new MsTranslator({api_key: translatorKey}, true);
-        this.translator.translateArrayAsync = denodeify(this.translator, this.translator.translateArray);
+    public constructor(translatorKey: string, protected nativeLanguages: string[], noTranslatePatterns: Set<string>, getUserLanguage?: (c: BotContext) => string, setUserLanguage?: (context: BotContext) => Promise<boolean>) {
```

Comment:
>translatorKey: string, protected nativeLanguages: string[], noTranslatePatterns: Set<string>, getUserLanguage?: (c: BotContext) => string, setUserLanguage?: (context: BotContext) => Promise<boolean> [](start = 23, length = 198)

Since you have so many config params you should consider taking a settings object to configure everything. That's more the JavaScript way.